Tasks/AzureRmWebAppDeployment/deployment/AzureRmWebAppDeploymentProvider.ts

```diff
@@ -0,0 +1,80 @@
+import { AzureAppServiceUtility } from '../operations/AzureAppServiceUtility';
+import tl = require('vsts-task-lib/task');
+import { FileTransformsUtility } from '../operations/FileTransformsUtility';
+import { DeployWar } from '../operations/WarDeploymentUtilities';
+import * as ParameterParser from '../operations/parameterparser'
+import { addReleaseAnnotation } from '../operations/ReleaseAnnotationUtility';
+import * as Constant from '../operations/Constants';
+
+var packageUtility = require('webdeployment-common/packageUtility.js');
+var deployUtility = require('webdeployment-common/utility.js');
+var msDeploy = require('webdeployment-common/deployusingmsdeploy.js');
+
+export class AzureRmWebAppDeploymentProvider implements IWebAppDeploymentProvider{
+    protected taskParams:TaskParameters;
+    protected appService: AzureAppService;
+    protected kuduService: Kudu;
+    protected appServiceUtility: AzureAppServiceUtility;
+    protected kuduServiceUtility: KuduServiceUtility;
+    protected virtualApplicationPath: string;
+    protected azureEndpoint: AzureEndpoint;
+    protected activeDeploymentID;
+
+    constructor(taskParams: TaskParameters) {
+        this.taskParams = taskParams;
+    }
+
+    public async PredeploymentStep() {
+        this.azureEndpoint = await new AzureRMEndpoint(this.taskParams.connectedServiceName).getEndpoint();
+        console.log(tl.loc('GotconnectiondetailsforazureRMWebApp0', this.taskParams.WebAppName));
+        if(!this.taskParams.DeployToSlotOrASEFlag) {
+            this.taskParams.ResourceGroupName = await AzureResourceFilterUtility.getResourceGroupName(this.azureEndpoint, this.taskParams.WebAppName);
+        }
+
+        tl.debug(`Resource Group: ${this.taskParams.ResourceGroupName}`);
+        this.appService = new AzureAppService(this.azureEndpoint, this.taskParams.ResourceGroupName, this.taskParams.WebAppName, 
+            this.taskParams.SlotName, this.taskParams.WebAppKind);
+        this.appServiceUtility = new AzureAppServiceUtility(this.appService);
+
+        await this.appServiceUtility.pingApplication();
+        this.kuduService = await this.appServiceUtility.getKuduService();
+        this.kuduServiceUtility = new KuduServiceUtility(this.kuduService);
+        tl.setVariable('AppServiceApplicationUrl', await this.appServiceUtility.getApplicationURL(!this.taskParams.isLinuxApp 
+            ? this.taskParams.VirtualApplication : null));
+    }
+
+    public async DeployWebAppStep() {}
+
+    public async UpdateDeploymentStatus(isDeploymentSuccess: boolean) {
+        if(this.kuduServiceUtility) {
+            await addReleaseAnnotation(this.azureEndpoint, this.appService, isDeploymentSuccess);
+            this.activeDeploymentID = await this.kuduServiceUtility.updateDeploymentStatus(isDeploymentSuccess, null, {'type': 'Deployment', slotName: this.appService.getSlot()});
+        }
+    }
+
+    protected async UpdateWebAppSettings() {
```

Comment:
method name is accurate based on what the function is doing

Reply:
Changed to PostDeploymentStep